java/core/src/main/scala/io/pdal/PointCloud.scala

```diff
@@ -0,0 +1,101 @@
+package io.pdal
+
+import java.nio.ByteBuffer
+import java.util
+
+import scala.collection.JavaConversions._
+
+/**
+  * PointCloud abstraction to work with packed point(s) in JVM memory.
+  * SizedDimType contains size and offset for a particular packed point with the current set of dims.
+  **/
+case class PointCloud(bytes: Array[Byte], dimTypes: util.Map[String, SizedDimType], metadata: String = "", schema: String = "") {
+  val pointSize: Long = dimTypes.values.map(_.size).sum
```

Comment:
I think it makes sense to cast it down on the translation, or else clients will always have to translate it themselves (and I can't see wanting this to be a long)